qirlib/src/generation/qir/instructions.rs

```diff
@@ -303,3 +234,45 @@ fn emit_if<'ctx>(
     emit_block(else_block, else_insts);
     generator.builder.position_at_end(continue_block);
 }
+
+fn get_value<'ctx>(
+    generator: &CodeGenerator<'ctx>,
+    env: &Environment<'ctx>,
+    value: &Value,
+) -> BasicMetadataValueEnum<'ctx> {
+    match value {
+        Value::Int(i) => generator
+            .context
+            .custom_width_int_type(i.width())
+            .const_int(i.value(), false)
+            .into(),
+        &Value::Double(d) => generator.f64_to_f64(d),
+        &Value::Qubit(id) => get_qubit(generator, id).into(),
+        &Value::Result(id) => get_result(generator, id).into(),
+        &Value::Variable(v) => env
+            .variable(v)
+            .unwrap_or_else(|| panic!("Variable {:?} not found.", v))
+            .into(),
+    }
+}
+
+fn get_qubit<'ctx>(generator: &CodeGenerator<'ctx>, id: u64) -> PointerValue<'ctx> {
+    let value = generator.u64_to_i64(id).into_int_value();
+    let ty = generator.qubit_type().ptr_type(AddressSpace::Generic);
+    generator.builder.build_int_to_ptr(value, ty, "")
+}
+
+fn get_result<'ctx>(generator: &CodeGenerator<'ctx>, id: u64) -> PointerValue<'ctx> {
+    let value = generator.u64_to_i64(id).into_int_value();
+    let ty = generator.result_type().ptr_type(AddressSpace::Generic);
+    generator.builder.build_int_to_ptr(value, ty, "")
+}
```

Comment:
It's allowed to use static qubit and result IDs that are outside the range declared by the `requiredQubits` and `requiredResults` attributes. I'm not sure if we should stop people from doing that.

Reply:
Based on our conversation, it is apparent that the attributes are set directly by the user without any checks, and get overwritten by processing in QAT anyway, so we should not worry about emitting these at all. They code related to qubit/result count attribute handling can be removed.

Reply:
Sounds good, I prefer removing the attributes too. FYI @idavis

Reply:
@swernli @samarsha FYI, for the future: https://github.com/qir-alliance/pyqir/issues/143